chore: add db management scripts and logging improvements (centralized logging)

GHalfbbt opened the pull request:
### Summary

This PR introduces multiple improvements to simplify database management and improve logging and testing workflows:

-  Added Django management commands for the clean, populate and rebuild database:
  - `clean_db` – to wipe all non-system tables
  - `faker_populate_db` – to seed the database with fake data using Faker
  - `rebuild_db` – unified script that cleans and repopulates the DB

-  Enhanced the `Logger` class:
  - Now supports subclass usage
  - Creates logs per script type (e.g., `tests.log`, `populate_db.log`, etc.)

-  Updated Donor model to use valid `choices` for "type" field

-  Fixed donor test payloads to use valid values for "type" field (e.g., `"individual" or "institución"`)

-  Updated `README.md`:
  - Added instructions for running scripts and tests
  - Linked database schema and usage

---

### Notes
- All logs are automatically written to `/logs` folder based on context
- All test output is also saved in `tests.log`

GHalfbbt committed:
chore: add db management scripts and logging improvements

osrodgon reviewed (commented):
No eniendo muy bien lo que has hecho con la clase Logger. He ejecutado algunos comandos y el log de tests y el log de la aplicacion son el mismo, con lo que no veo el motivo por el que has decidido separarlos.

Por otro lado, no es necesario datos para los tests ni modificar los test. Los mismos tests se encargan de generar los datos que necesitan.

Lo hablamos cuando pasemos a las salas.

osrodgon commented:
Bueno, creo que ya he visto el problema. Vamos por partes:

__Usar \_logger_app en lugar de \_\_logger_app__
He cambiado la variable de protegida a privada y ha funcionado sin ningun problema. Prefiero que sea privada.

__Uso de la clase Logger__

He visto que creas una instancia de la clase para pasarle el nombre del fichero log a usar. El problema principal es que la clase esra pensada para ser una clase abstracta, con lo que deberias de heredar la clase en lugar de instanciarla. Si la heredas, puede dar problemas, asi que lo mejor es dejarlo como lo tienes (se puede llamar al contructor de la superclase, pero no merece la pena)

__Cambios a la configuracion del log en settings__
Esto choca directamente con la PR de Ciprian, que mueve esos settins a variables de entorno. Habria que comprobar que todo es correcto a la hora de migrar.

Y una cosa mas: Creo que deberiamos eliminar el test.log. No tiene sentido separlo del general puesto que los tests solo se ejecutan en desarrollo y nunca en produccion, ademas de no dar mucha informacion de donde se esta generando el mensaje al eleminar el nombre de la clase por "test".

osrodgon commented on food_pantry/common/logger.py:
```diff
@@ -1,5 +1,117 @@
+
+    def __init__(self, log_name=None):
+        # Initialize logger with optional custom name
+        self._logger_app = self.init_log(log_name)
```
Renombra de _loger_app a __loger_app

El constructor deberia de ser siempre (por convencion) el primer metodo de una clase. Muevelo, por favor.

GHalfbbt replied:
Actualizado logger.py. 
Se renombra de _loger_app a __loger_app para que sus métodos sigan siendo privados.

osrodgon commented on food_pantry/fixtures/management/commands/clean_db.py:
```diff
@@ -0,0 +1,128 @@
+
+
+# abstract class for logging
+class Command(BaseCommand):
```
Aqui, si quieres usar la clase como esta concebida, deberia de ser (el orden es importante):

class Command(Logger, BaseCommand)

y luego crear el constructor:

def __init__(self):
     super().__init__('clean_db')

Con esto ya no te hace falta declarar el logger en la linea 26 y usas el log llamando a self:

self.warning("My warning message.")

OJO!!! Esto es totalmente opcional, es a titulo informativo

GHalfbbt replied:
No lo he modificado. Se ha dejado igual.

osrodgon commented on food_pantry/fixtures/management/commands/faker_populate_db.py:
Mismo comentario que para clean_db

GHalfbbt replied:
No se ha modificado. Se ha dejado igual.

osrodgon commented on food_pantry/fixtures/management/commands/rebuild_db.py:
Mismo comentario que para clean_db

GHalfbbt replied:
No se ha modificado. Se ha dejado igual.

osrodgon commented on food_pantry/food_pantry/settings.py:
```diff
@@ -12,14 +12,26 @@
 # Logger settings
 LOG_SETTINGS={
-    "file_name": "food_pantry",
+    #"file_name": "food_pantry",
```
Todos estos cambios chocan con la PR de Ciprian. Esperemos que no haya problemas (dependera de quien haga el merge primero)

GHalfbbt replied:
settings.py adaptado para que coja las variables de entorno definidas en ".env".

Modificado también, para que los logs de los tests, se escriban en logs/food_pantry.log

osrodgon commented on readme.md:
```diff
@@ -67,6 +67,14 @@ food_pantry/
 ├── venv/                      Entorno virtual (no incluido en el repositorio)
 ├── requirements.txt           Lista de dependencias
+├── common
+|     ├── test/                Lógica implementación de logs del sistema (logger)
```
En test estan los test unitarios, no la logica del sistema de logs. La logica esta en el fichero logger.py

GHalfbbt replied:
Actualizado en el README carpeta test y logger.py (que contiene realmente la lógica de los tests)

osrodgon commented on readme.md:
````diff
@@ -115,10 +123,55 @@ Una descripción completa de las entidades, atributos, relaciones y restriccione
+
+Para ejecutar todos los tests de la aplicación y ver el resumen detallado, se han definido pruebas automatizadas con pytest y pytest-django.:
+
+```
````
Quizas podrias añadir aqui tambien la opcion de lanzar pytest con coverage (pytest --cov)

GHalfbbt replied:
Añadido al README ejecución de test con "pytest --cov"

GHalfbbt committed:
Refactor: Renamed _logger_app to __logger_app for encapsulation

GHalfbbt committed:
Merge branch 'Develop' of https://github.com/Factoria-F5-madrid/proyecto2_grupo1_CRUD into chore_scripts-populate-db-utils

GHalfbbt committed:
Resolve merge with Develop: .env, settings, requirements, logger, test fixtures

GHalfbbt commented:
This update completes the integration of recent changes from the Develop branch and includes the following:

 **Merge Conflict Resolution**

-     Merged .env support into settings.py for environment-based logging configuration

-     Resolved merge conflicts in:

        settings.py

        requirements.txt

        logger.py

        Test fixtures and deleted legacy fixtures directory

**Logger changes:**

-     Logger now reads configuration from environment variables (.env)

-     Test logs are now redirected to the default application log file (logs/food_pantry.log) instead of logs/tests.log

 **Fixture & Script Enhancements**

-     Updated and tested clean_db.py, faker_populate_db.py, and unified rebuild_db.py

-     Scripts run cleanly and follow logging conventions

 **Testing & Coverage**

-     All unit tests passing

-     Test coverage verified using pytest --cov

 **README updated**
 
 cleaner_db.py, faker_populate_db.py y rebuild_db.py --> Files not modified.
